rewrite-java/src/main/java/org/openrewrite/java/ChangePackage.java

```diff
@@ -113,14 +116,45 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
                     }
                     stopAfterPreVisit();
                 }
+                if (tree instanceof SourceFileWithTypeReferences) {
+                    SourceFileWithTypeReferences cu = (SourceFileWithTypeReferences) requireNonNull(tree);
+                    for (TypeReference ref : cu.getTypeReferences().getTypeReferences()) {
+                        if (ref.getName().startsWith(oldPackageName)) {
+                            return SearchResult.found(cu);
+                        }
+                    }
+                }
                 return super.preVisit(tree, ctx);
             }
         };
 
-        return Preconditions.check(condition, new ChangePackageVisitor());
+        return Preconditions.check(condition, new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public boolean isAcceptable(SourceFile sourceFile, ExecutionContext ctx) {
+                return sourceFile instanceof JavaSourceFile || sourceFile instanceof SourceFileWithTypeReferences;
+            }
+
+            @Override
+            public @Nullable Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
+                if (tree instanceof JavaSourceFile) {
+                    return new JavaChangePackageVisitor().visit(tree, ctx);
+                } else if (tree instanceof SourceFileWithTypeReferences) {
+                    SourceFileWithTypeReferences sourceFile = (SourceFileWithTypeReferences) tree;
+                    SourceFileWithTypeReferences.TypeReferences typeReferences = sourceFile.getTypeReferences();
+                    Set<TypeReference> matches = new HashSet<>();
+                    for (TypeReference ref : typeReferences.getTypeReferences()) {
+                        if (ref.getName().startsWith(oldPackageName)) {
```

Comment:
Do we need to already here check for `recursive` and make sure the ref also has a dot after the package name?